drivers/npmx/npmx_driver.c

```diff
@@ -15,7 +15,7 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(NPMX, CONFIG_NPMX_LOG_LEVEL);
 
-#define DT_DRV_COMPAT nordic_npm1300_eng_c
+#define DT_DRV_COMPAT nordic_npm1300
```

Comment:
Not sure if this will work, it could conflict with upstream binding.
If problematic, it could be renamed to `nordic_npmx_npm1300` or something similar.